### PR TITLE
Set as wallpaper/contact-photo/etc.

### PR DESCRIPTION
**Pull request by AndyScherzinger**
revived oC 1240 by @tobiasKaminsky 

> Ability to set a picture as wallpaper, facebook, contact, whatsapp, etc.

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
The set as doesn't work (for me). Simply choose a picture and click on menu item "set picture as" will give you a choice of app. I chose Google Photos (means Wallpaper) and then Google Photos crahed with the following stack trace:
```
08-04 14:26:13.382 29176-29176/? E/AndroidRuntime: FATAL EXCEPTION: main
Process: com.google.android.apps.photos, PID: 29176
android.os.FileUriExposedException: file:///storage/emulated/0/nextcloud/andy%40********/IMG_20150628_140508.jpg exposed beyond app through Intent.getData()
    at android.os.StrictMode.onFileUriExposed(StrictMode.java:1799)
    at android.net.Uri.checkFileUriExposed(Uri.java:2346)
    at android.content.Intent.prepareToLeaveProcess(Intent.java:8933)
    at android.content.Intent.prepareToLeaveProcess(Intent.java:8894)
    at android.app.Instrumentation.execStartActivity(Instrumentation.java:1517)
    at android.app.Activity.startActivityForResult(Activity.java:4224)
    at cx.startActivityForResult(PG:48)
    at de.startActivityForResult(PG:75)
    at ujz.startActivityForResult(PG:199)
    at android.app.Activity.startActivityForResult(Activity.java:4183)
    at de.startActivityForResult(PG:871)
    at ujz.startActivityForResult(PG:190)
    at android.app.Activity.startActivity(Activity.java:4507)
    at ujz.startActivity(PG:164)
    at android.app.Activity.startActivity(Activity.java:4475)
    at ujz.startActivity(PG:155)
    at mcv.a(PG:1123)
    at sjh.a(PG:434)
    at sjw.a(PG:323)
    at sjx.run(PG:1027)
    at vnr.run(PG:95)
    at sju.run(PG:1291)
    at vnr.run(PG:95)
    at android.os.Handler.handleCallback(Handler.java:751)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6077)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
```

File: res/values/setup.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
     migrate accounts when the app is updated -->
     <string name="authority">org.nextcloud</string>	<!-- better if was the app package with ".provider" appended ; it identifies the provider -->
     <string name="document_provider_authority">org.nextcloud.documents</string>
+    <string name="authorityCache">org.nextcloud.imageCache.provider</string>
```
Comment (AndyScherzinger) [inline res/values/setup.xml]: @tobiasKaminsky is this correct and also is this needed? I couldn't find any reference to this anywhere.

**Comment by tobiasKaminsky** [comment]
You are using Android N, right? ;)
Then this is the problem and we need to solve it this way: http://stackoverflow.com/a/38925453/5074333

**Comment by AndyScherzinger** [comment]
I am on Android N, yes. Nice catch @tobiasKaminsky !!! :rocket:

**Comment by tobiasKaminsky** [comment]
Finally! :tada: 

Please have look :+1:

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger @mario please approve

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky tested and found one minor UX related issue:

Mark a not downloaded picture in the file list, the menu contains "set picture as", while clicking it doesn't have any action. So either this would need to trigger a download upfront or the item should be hidden from the menu imho.

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger great that you have found it! 
I have fixed it: first the file will get downloaded. This is the same behaviour as in "send file".

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky fix tested successfully while I tried another thing which we should fix. Since SVG is an image by mimeType "set as picture" is also offered and Android actually offers Contact and background image while choosing either of them simply fails silently returning back to our file list. So it is not an issue of our app but of Android so I's still say we might want to hide the menu item for SVG. What do you think?

**Comment by tobiasKaminsky** [comment]
:+1: disabled it for svg images

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @mario fixed some lint warnings and retested the PR, so please re-approve and merge 🚀

**Comment by tobiasKaminsky** [comment]
:+1:

**Comment by mario** [comment]
👍

**Pull request merged**